contracts/asset/src/execute/listing.rs

```diff
@@ -0,0 +1,113 @@
+use cosmwasm_std::{Coin, CustomMsg, DepsMut, Env, MessageInfo, Response};
+use cw721::traits::Cw721State;
+
+use crate::{
+    error::ContractError,
+    state::{AssetConfig, ListingInfo, Reserve},
+};
+
+use super::permissions::check_can_list;
+
+pub fn list<TNftExtension, TCustomResponseMsg>(
+    deps: DepsMut,
+    env: &Env,
+    info: &MessageInfo,
+    id: String,
+    price: Coin,
+    reservation: Option<Reserve>,
+    marketplace_fee_bps: Option<u16>,
+    marketplace_fee_recipient: Option<String>,
+) -> Result<Response<TCustomResponseMsg>, ContractError>
+where
+    TNftExtension: Cw721State,
+    TCustomResponseMsg: CustomMsg,
+{
+    let asset_config = AssetConfig::<TNftExtension>::default();
+    // make sure the caller is the owner of the token
+    let nft_info = asset_config.cw721_config.nft_info.load(deps.storage, &id)?;
+    // check if we can list the asset
+    check_can_list(deps.as_ref(), env, info.sender.as_ref(), &nft_info)?;
+    // make sure the price is greater than zero
+    if price.amount.is_zero() {
+        return Err(ContractError::InvalidListingPrice {
+            price: price.amount.u128(),
+        });
+    }
+
+    let (validated_marketplace_fee_bps, validated_marketplace_fee_recipient) =
```

Comment:
Seems like these marketplace values can be set by the lister. How do they interact with the marketplace? Why would someone listing an asset set them?

Reply:
an individual listing an item would not set this. This is strictly for scenarios where marketplaces are the ones listing and want to set a fee